mleap-tensorflow/src/main/scala/ml/combust/mleap/tensorflow/TensorflowModel.scala

```diff
@@ -64,26 +70,34 @@ case class TensorflowModel(@transient var graph: Option[tensorflow.Graph] = None
   }
 
   private def withSession[T](f: (tensorflow.Session) => T): T = {
-    val g = graph match {
-      case Some(gg) => gg
-      case _ => { // can also be null at deserialization time, not just empty
-        val gg = new tensorflow.Graph()
-        val graphDef = GraphDef.parseFrom(graphBytes)
-        gg.importGraphDef(graphDef)
-        graph = Some(gg)
-        graph.get
+    val (s,g) = (session, graph) match {
+      case (Some(sess), Some(gg)) => (sess, gg)
+      case _ => format match {
+        case Some("graph") | None => getSessionFromFrozenGraph
+        case Some("saved_model") => getSessionFromSavedModel
+        case _ =>  throw new RuntimeException("Only graph and saved_model are supported")
       }
     }
+    session = Some(s)
+    graph = Some(g)
+    f(s)
+  }
 
-    val s = session match {
-      case Some(sess) => sess
-      case _ => { // can also be null at deserialization time, not just empty
-        session = Some(new tensorflow.Session(g))
-        session.get
-      }
-    }
+  private def getSessionFromFrozenGraph: (tensorflow.Session, tensorflow.Graph) = {
+    val graph = new tensorflow.Graph()
```

Comment:
does this have a namespace collision with the class `graph` attribute?  I think no?  But might be better to pick another name?

Reply:
will do